rio_tiler/io/cogeo.py

```diff
@@ -135,6 +138,19 @@ def maxzoom(self) -> int:
             self._get_zooms()
         return self._maxzoom
 
+    def _update_kwargs(self, kwargs: Dict):
+        nodata = kwargs.get("nodata", self.nodata)
+        if nodata is not None:
+            kwargs["nodata"] = nodata
+
+        vrt_options = kwargs.get("vrt_options", self.vrt_options)
+        if vrt_options is not None:
+            kwargs["vrt_options"] = vrt_options
+
+        unscale = kwargs.get("unscale", self.unscale)
+        if unscale is not None:
+            kwargs["unscale"] = unscale
+
```

Comment:
Please, if there is a better way, feel free to edit this.

Reply:
I think we can use `__post_init__` (https://docs.python.org/3/library/dataclasses.html#post-init-processing)